processing/src/main/java/org/apache/druid/query/groupby/epinephelinae/BufferArrayGrouper.java

```diff
@@ -69,12 +69,20 @@
   @Nullable
   private int[] vAggregationRows = null;
 
-  static long requiredBufferCapacity(
-      int cardinality,
-      AggregatorFactory[] aggregatorFactories
-  )
+  /**
+   * Computes required buffer capacity for a grouping key of the given cardinaltiy and aggregatorFactories.
+   * This method assumes that the given cardinality doesn't count nulls.
+   *
+   * Returns -1 if it cardinality + 1 (for null) = Integer.MAX_VALUE. Returns computed required buffer capacity
```

Comment:
I think the comment here should have `> Integer.MAX_VALUE` instead

Reply:
Oops, thanks!